### PR TITLE
Fix 500 error when trying to cancel a booking that was already cancelled.

### PR DESCRIPTION
**Pull request by joshsny**
This PR fixes issue #387.

 - Cancelling an event that was already cancelled returned a 500 error, instead, it now displays the following page:

Screenshot 2021-08-10 at 10 56 20

- The error page also no longer displays the Cancel and Reschedule buttons.

Unfortunately, I lumped this in with a previous PR #429, please feel free to remove the changes from the `pages/availability/troubleshoot.tsx` file.

### CODE REVIEW

File: pages/availability/troubleshoot.tsx [hunk 1]
```diff
@@ -30,23 +26,29 @@ export default function Troubleshoot({ user }) {
...
+      })
       .catch((e) => {
         console.error(e);
       });
   };
 
-  fetchAvailability(selectedDate);
+  useEffect(() => {
+    fetchAvailability(selectedDate);
+  });
```
Comment (Nico-J) [inline pages/availability/troubleshoot.tsx:47]: Regarding your fix for the API-flooding loop:
I tested your code (current state from your branch) and it doesn't fix it, at least for me - it still refreshes forever and floods the API. However, when watching `selectedDate` in the useEffect closure, it works perfectly for me. :)

```suggestion
  }, [selectedDate]);
```

**Review by Nico-J — CHANGES_REQUESTED** [review]
Hey Joshua,
Thank you very much for your PR. :) I've tested it and the "already canceled" message works fine. :+1: 
Though, the "API flooding" issue has not been resolved in my case. I've made a suggestion that fixes it for me. Could you include that in your PR?

In addition, the availability endpoint is currently broken in some cases. There's already another PR that is going to fix that. With both fixes combined, the troubleshooting page works like a charm:

image

**Comment by joshsny** [comment]
Great. Thanks for that suggestion @Nico-J :)

**Comment by joshsny** [comment]
Hi @Nico-J,

I found a bug in this PR when testing it again. The cancellation was failing when a booking had not been cancelled, but the event type had already been deleted.

I have gone ahead and fixed that now.

**Review by Nico-J — APPROVED** [review]
Perfect. :) From my side, it's all fine now. :)

**Pull request merged**